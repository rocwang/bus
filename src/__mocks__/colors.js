export default {
  oceanBlue: "#001120",
  green: "#8fc240",
  blue: "#00a3da",
  yellow: "#ffdb1b",
  red: "#da2f2d",
  orange: "#ea8c2e",
  black: "#000",
  grayA: "#333",
  grayB: "#666",
  grayC: "#999",
  grayD: "#ccc",
  white: "#fff",
};
