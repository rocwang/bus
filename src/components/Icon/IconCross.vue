<template>
  <path
    fill="none"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    d="M15 15L29 1 15 15 1 1l14 14zm0 0L1 29l14-14 14 14-14-14z"
  />
</template>

<script>
export default {
  name: "IconCross",
};
</script>
