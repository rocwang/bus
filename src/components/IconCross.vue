<template>
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30 30">
    <path
      fill="none"
      :stroke="color"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      d="M15 15L29 1 15 15 1 1l14 14zm0 0L1 29l14-14 14 14-14-14z"
    />
  </svg>
</template>

<script>
export default {
  name: "IconCross",
  inject: ["colors"],
  props: {
    color: {
      type: String,
      default() {
        return this.colors.yellow;
      }
    }
  }
};
</script>
