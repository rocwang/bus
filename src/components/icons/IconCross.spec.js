import { mount } from "@vue/test-utils";
import IconCross from "./IconCross";
import colors from "../__fixtures__/colors";

describe("IconCross.vue", () => {
  it("matches the snapshot", () => {
    const wrapper = mount(IconCross, {
      provide: {
        colors
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has a stroke color", () => {
    const wrapper = mount(IconCross, {
      provide: {
        colors
      },
      propsData: {
        color: colors.red
      }
    });
    expect(wrapper.find("path").attributes("stroke")).toBe(colors.red);
  });
});
