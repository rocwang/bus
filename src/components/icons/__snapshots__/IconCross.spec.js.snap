// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconCross.vue matches the snapshot 1`] = `
<svg
  viewBox="0 0 30 30"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M15 15L29 1 15 15 1 1l14 14zm0 0L1 29l14-14 14 14-14-14z"
    fill="none"
    stroke="#ffdb1b"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
  />
</svg>
`;
